_ml-commons-plugin/model-serving-framework.md

```diff
@@ -0,0 +1,250 @@
+---
+layout: default
+title: Model serving framework 
+has_children: false
+nav_order: 110
+---
+
+# Model serving framework
+
+ML commons supports the ability to serve custom models and use those models to make inferences. For those who wish to use train their models using a deep-learning framework, such as Pytorch, they can upload such models to OpenSearch in order to take advantage of OpenSearch resources.
+
+This page outlines the steps required to upload a custom model with the ML Commons plugin.
+
+## Prerequisites 
+
+To upload a custom model to OpenSearch, you'll need to prepare outside of your OpenSearch cluster. You can use a current model or train a new model depending on your needs.
+
+### Model format
+
+Although most ML models are written in Python, you'll need to export the model into a portable format. As of 2.4, OpenSearch only supports [torchscript](https://pytorch.org/docs/stable/jit.html).
```

Comment:
We only support text embedding model in 2.4.

Reply:
So torchscript is the only portable format and text embedding is the only model?

Reply:
> torchscript is the only portable format

Yes, correct

> text embedding is the only model

Text embedding is a class of model which will translate a sentence into a dense vector (you can think it as a list of float data). @dylan-tong-aws please chime-in if you have better explanation.

Reply:
That's explained partially in the Neural Search docs, https://github.com/opensearch-project/documentation-website/pull/1882/files#diff-6be3cf6f5a7bbb1e951a5fe96947aeed71ce22642dea61b91b1251de932011f7R24.

Reply:
Do we still want to include a note about text embedding here?

Reply:
Text embedding models can encode text into vectors that capture semantics enabling the use of OpenSearch's k-NN indexes to answer similarity queries.

Reply:
But are text embedding models the only model supported in the Model Framework @dylan-tong-aws?

Reply:
> Do we still want to include a note about text embedding here?

I think yes. No need to ask user read two places to learn the whole picture.

Reply:
> But are text embedding models the only model supported in the Model Framework @dylan-tong-aws?

Yes, I can confirm that.